manifests/plugin/java.pp

```diff
@@ -19,13 +19,23 @@
       }
     }
     if $java_home {
-      file { '/usr/lib64/libjvm.so':
-        ensure => 'link',
-        target => "${java_home}/jre/lib/amd64/server/libjvm.so",
+      # OpenJDK based Java distribution
+      exec { "/usr/bin/ln -s /usr/lib64/libjvm.so ${java_home}/jre/lib/server/libjvm.so":
```

Comment:
Why did you replace the file resource with an exec one? The file resource can handle symlinks.

Reply:
Indeed it can and I thought about it but how would you implement a conditional `File` resource which would create `/usr/lib64/libjvm.so` depending on certain conditions i.e. whether `${java_home}/jre/lib/server/libjvm.so` exists or `${java_home/jre/lib/amd64/server/libjvm.so`? Maybe I am missing something here but the `File` resource doesn't have an `onlyif` parameter and the only way AFAICT to do something like this is via an `exec` resource.